src/network/service.rs

```diff
@@ -334,7 +336,27 @@ where
         protocol::decode_storage_proof_response(&response).map_err(StorageProofRequestError::Decode)
     }
 
-    pub async fn announce_transaction(&self, transaction: Vec<u8>) {}
+    pub async fn announce_transaction(
+        &self,
+        target: &peer_id::PeerId,
+        chain_index: usize,
+        extrinsic: Vec<u8>,
+    ) -> Result<Vec<u8>, QueueNotificationError> {
+        let mut val = Vec::new();
+        val.extend_from_slice(util::encode_scale_compact_usize(1).as_ref());
+        val.extend(extrinsic.clone());
+        self.libp2p
+            .queue_notification(
+                &target,
+                chain_index * NOTIFICATIONS_PROTOCOLS_PER_CHAIN + 1,
+                val.clone(),
+            )
+            .await?;
+
+        let mut hash = blake2_rfc::blake2b::Blake2b::new(32);
+        hash.update(val.as_slice());
+        Ok(hash.finalize().as_bytes().to_vec())
```

Comment:
The hash returned by `author_submitExtrinsic` has nothing to do with networking. It should be done by the code that processes the `author_submitExtrinsic`, not here.